mesonbuild/compilers/cpp.py

```diff
@@ -76,6 +120,7 @@ def __init__(self, exelist, version, cltype, is_cross, exe_wrapper=None, **kwarg
         self.warn_args = {'1': default_warn_args,
                           '2': default_warn_args + ['-Wextra'],
                           '3': default_warn_args + ['-Wextra', '-Wpedantic']}
+        self.cpp_std_args = {}
```

Comment:
Instead of manually managing the cache here and for `GnuCPPCompiler`, we should use the `@functools.lru_cache()` decorator on `_find_best_cpp_std()`.

Reply:
Done